pkg/api/repository_package_groups.go

```diff
@@ -0,0 +1,31 @@
+package api
+
+import "github.com/lib/pq"
+
+type RepositoryPackageGroup struct {
+	UUID        string   `json:"uuid"`        // Identifier of the package group
+	ID          string   `json:"id"`          // The package group ID
+	Name        string   `json:"name"`        // The package group name
+	Description string   `json:"description"` // The package group description
+	PackageList []string `json:"packagelist"` // The list of packages in the package group
+}
+
+type RepositoryPackageGroupCollectionResponse struct {
+	Data  []RepositoryPackageGroup `json:"data"`  // List of package groups
+	Meta  ResponseMetadata         `json:"meta"`  // Metadata about the request
+	Links Links                    `json:"links"` // Links to other pages of results
+}
+
+type SearchPackageGroupResponse struct {
+	PackageGroupName string         `json:"package_group_name"`            // Package group found
+	Description      string         `json:"description"`                   // Description of the package group found
+	PackageList      pq.StringArray `json:"package_list" gorm:"type:text"` // Package list of the package group found
+}
```

Comment:
You may actually be able to do a shared response struct if you do something like this

```go
type SearchContentResponse struct {
  Name string
  Description string
}

type SearchPackageGroupResponse struct {
  *SearchContentResponse
  PackageList []string
}
```

This would share the fields of the first struct with the second struct. Not 100% sure if SearchContentResponse actually needs to be a pointer, but I think it would be.

What do you think?

Reply:
i like this! do you think i should use this base search response for rpms as well and have the base just include a name?

Reply:
it looks like name and description are common to all 3, so I would use the base for rpm and environments and then just add package list for package groups

Reply:
since we discussed staying consistent with the xml files (rpms use summary and groups / envs use description) we decided to keep this struct as is for now